### PR TITLE
fix: change list widget api

### LINKED ISSUE #18287: Task:  Change List widget's APIs
`gridGap`      --> `rowGap`
`rowIndex`  ---> `viewIndex` in MetaWidgetGenerator
`index` ---> `rowIndex` in MetaWidgetGenerator

```
selectedItemIndex       -> selectedRowIndex       -> selectedItemIndex
selectedItem            -> selectedItem           -> selectedItem
currentSelectedViewItem -> selectedRow            -> selectedItemView
currentViewItems        -> currentViewRows        -> currentItemsView
                        -> triggeredRowIndex      -> triggeredItemIndex
                        -> triggeredRow           -> triggeredItemView

```

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Updated some of the List Widget V2 Apis.

`gridGap`      --> `rowGap`
`rowIndex`  ---> `viewIndex` in MetaWidgetGenerator
`index` ---> `rowIndex` in MetaWidgetGenerator

```
selectedItemIndex       -> selectedRowIndex       -> selectedItemIndex
selectedItem            -> selectedItem           -> selectedItem
currentSelectedViewItem -> selectedRow            -> selectedItemView
currentViewItems        -> currentViewRows        -> currentItemsView
                        -> triggeredRowIndex      -> triggeredItemIndex
                        -> triggeredRow           -> triggeredItemView
currentRow              -> currentView            -> currentView
```
There are tests in place to catch places I might have omitted.

Fixes #18287

## Type of change
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export type GeneratorOptions = {
   currTemplateWidgets: TemplateWidgets;
   prevTemplateWidgets?: TemplateWidgets;
   data: Record<string, unknown>[];
-  gridGap: number;
+  rowGap: number;
```
Comment (ashit-rath) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:44]: I think this should be `itemGap`

Comment (Tooluloope) [reply app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:44]: Got it

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**